minidump-unwind/src/amd64.rs

```diff
@@ -138,37 +127,70 @@ where
         // drowning the rest of the code in checked_add.
         return None;
     }
-    let caller_ip = stack_memory.get_memory_at_address(last_bp + POINTER_WIDTH)?;
-    let caller_bp = stack_memory.get_memory_at_address(last_bp)?;
-    let caller_sp = last_bp + POINTER_WIDTH * 2;
 
-    // If the recovered ip is not a canonical address it can't be
-    // the return address, so bp must not have been a frame pointer.
+    let resolve = |offset_max_scan, offset_step| -> Option<(u64, u64, u64)> {
+        for offset in 0..=offset_max_scan {
+            let offset = offset * offset_step;
```

Comment:
You could also make this a function of just a single `offset` and move the iteration logic below to the Windows case (e.g. with `find_map`).

Reply:
the `find_map` wouldn't allow for an early exit when the `last_bp + offset` is going past the stack.

Reply:
Ah, you mean we would always iterate through the entire list even if we already know we're out of bounds? That's a fair point.